app/views/smart_proxies/plugins/puppet/_puppet_ca.html.erb

```diff
@@ -0,0 +1,43 @@
+<div class="row">
+  <div class="col-md-12">
+    <div class="row">
+      <h4><%= _('Puppet CA') %></h4>
+    </div>
+    <%= render :partial => 'smart_proxies/plugins/plugin_version', :locals => {:feature => 'puppetca', :proxy => @smart_proxy } %>
+    <div class="row">
+      <div class="col-md-4">
+        <%= _('Hosts count') %>
+      </div>
+      <div class="col-md-8">
+        <%= link_to(@smart_proxy.puppet_ca_hosts.count, hosts_path(:search => "puppet_ca = #{@smart_proxy.name}")) %>
+      </div>
+    </div>
+    <% %w(valid pending revoked).each do |state| %>
+      <div class="row">
+        <div class="col-md-4">
+          <%= _('%s certificates') % state.capitalize %>
+        </div>
+        <div class="col-md-8">
+          <% state_count =  SmartProxies::PuppetCA.find_by_state(@smart_proxy, state).count rescue _('Error getting data for %s certificates') % state %>
```

Comment:
this will "eat" the exception. IMHO you need to change it to a service class, that can later on be used in the api too.